packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1106,6 +1135,20 @@ export class ContainerRuntime
 		this.maxOpsSinceLastSummary = this.getMaxOpsSinceLastSummary();
 		this.initialSummarizerDelayMs = this.getInitialSummarizerDelayMs();
 
+		// Enabling the compressor is a one-way operation. Once it's enabled for a document
+		// it is persisted in the container's metadata.
+		const compressorEnabled =
+			metadata?.idCompressorEnabled ??
+			this.mc.config.getBoolean("Fluid.ContainerRuntime.IdCompressorEnabled") ??
```

Comment:
Should the feature flag override the value in metadata? That way, if something goes wrong, you can disable id compression for documents that has it enabled. I am not really sure what is the right approach because it varies from feature to feature.

Reply:
I thought about this for a while when I wrote this. I think it should be this way until the compressor is enabled by default for all containers. Containers that turn it on before then will be doing so because they are specifically using the compressor and turning it off would result in not being able to allocate ids or using ones they've already allocated. In the future, assuming this is turned on by default, I think the feature flag can override the metadata so that we have an emergency off switch for containers that aren't explicitly reliant on the compressor.